### ISSUE TITLE
Visual Debug view w/registers - Color change of modified registers sometimes stops updating

### ISSUE THREAD

**Issue by pgrohe** [radare/radare2#5490]
When debugging in visual disassembly with registers view  the color change of modified registers will stop updating / lose synch with the instructions being stepped through.  

Registers will retain their modified color and never return to their normal color state even when stepping over instructions that do not affect them.

This seems to occur after stepping through some instructions initially, continuing to a breakpoint and then stepping through some more instructions.
![radare2-modified-registers](https://cloud.githubusercontent.com/assets/359095/17644591/e8773aaa-6158-11e6-8185-7421c6619f78.png)
![radare2-modified-registers-broken](https://cloud.githubusercontent.com/assets/359095/17644592/e89e428a-6158-11e6-9b8d-61c4530011d2.png)

Steps to Reproduce
----------------------------
1 - Get source files for the Radare2 Explorations book here:  https://github.com/monosource/radare2-explorations-binaries
2 - Build the patchme executable found in the director tut1-patchme directory
3 - Debug the patchme in radare with the command:  r2 -Ad ./patchme
4 - Switch to visual mode disassembly w/registers by typing: V, p, p
5 - Step over first few instructions in /lib32/ld-2.23.so, note register coloration updating normally (on default color scheme changed registers are the blue-ish color, unchanged are the standard green terminal color)
6 - Seek to main by pressing 'o' in visual mode and typing 'main' and then Enter
7 - Press 'b' to set a breakpoint at start of main, then 'F9' to continue to break
8 - Note register colors, many are in blue 'changed' color
9 - Step over several instructions using 's'.  Note that the registers remain blue even though the previously stepped over instruction did not modify them.  Registers should lose their 'modified' coloration after instruction where they are not modified.

**Comment by Maijin** [comment]
Hello,

Ensure you are using radare2 from git, if you're unsure paste output of `r2 -v` here.
To install radare2 from git, first uninstall your version of radare2 and clean your distro. Then use `git clone https://github.com/radare/radare2 && cd radare2 && ./sys/install.sh`, verify your version and check if there is no error using `r2 -v`.

**Comment by Maijin** [comment]
Can you also create an asciinema

**Comment by pgrohe** [comment]
Hi,

Yup, got radare2 from git. Update a couple of days ago. Version string is:
radare2 0.10.5-git 12047 @ linux-x86-64 git.0.10.4-termux4-37-ga7da94b
commit: a7da94ba336aa954d4a85639cb7226c9eaf44984 build: 2016-08-11

Shall I update to latest and try or was there anything you wanted me to check in the current state?

asciinema is here:
https://asciinema.org/a/4b3k7bb5agd5emay766gepp27

**Comment by radare** [comment]
I think this happens because the regstate is reset when performing a dc. Can you confirm if this happens when using dcu?

**Comment by pgrohe** [comment]
Still occurs with when doing 'dcu main'.  

Once you hit the break most of the registers are already the modified color.  They don't reset to normal color as you step.  Registers that weren't already in the modified color will still change to it if an instructions affects them though.

**Comment by radare** [comment]
so its a problem with breakpoints?

**Comment by pgrohe** [comment]
Yes.  Seems to be.  While looking into this I've never encountered it when single stepping right from the beginning of a debug session.  Always occurs after continuing to a break in one way or another.

I also tried changing dbg.bep to 'main' instead of 'loader' and then reloading the file with 'oo'. The issue occurs in that case as well.  The moment execution is stopped at the start of main the registers are almost all.
